### DISCUSSION TITLE
OPC discovery takes to long and gives an error

### DISCUSSION THREAD

**Discussion by tayfun-celik** [apache/streampipes#3134] [Q&A]
Hello All,

I am new in streampipe. I connected OPC UA server and forward this information to postgre. And it was succesful. Streampipes and opc server located different site and I made connection with VPN tunnel. So when streampipes discover tags in OPC UA it takes time and show error page in most cases. So is it possible manually add these tags as OPC adapter? Is there any import file for that?

Best Regards

Tayfun

**Comment by tenthe** [comment]
Hi @tayfun-celik,

Thank you for initiating this discussion.
We are currently working on an alternative approach to manually selecting the nodes in a text editor (#3112).
This should help users who already know which nodes to add.

Below is a screenshot of a first version of the text editor, as an alternative to the tree editor.


<img width="1335" alt="grafik" src="https://github.com/user-attachments/assets/8dcf9696-12da-4173-9dec-01037003a742">

**Reply by tayfun-celik** [reply]
Hello Tenthe,

Thank you for answer. Is there any planned date for this future?

Yesterday I try to add many tags to opc adapter. When I added 750 tag there is no problem. When number of tags more than 900 the problem is occurs. Streampipes does not give permission to do this and gives an error. Is there any known limit for adding OPC tags? For this reason I try to divide different OPC adapters in configuration. This time I observed more than two opc adapter is not possible to start. Is there any known limit for this?

Best Regards,

Tayfun CELIK

**Reply by tenthe** [reply]
Hi,

The feature is currently being developed in #3138 and will be part of the next release.

Regarding the number of connected nodes, there is no known limitation. You should also be able to start multiple OPC UA adapters. If you encounter any issues, it would be helpful to review the logs of the extensions service for any error messages.
Additionally, more information about your StreamPieps setup (which version did you start) and the OPC UA server would be useful for further diagnosis.

**Reply by tayfun-celik** [reply]
Hello @tenthe ,
My version is 0.95.1. And my OPC server is hardware based weintek cmt-g01 device. For more information please look this link https://www.weintek.com/globalw/Product/Product_speccMTG.aspx . My setup is ubuntu system and docker container. Where I can get log files of streampipes for further analyses?

Best Regards,

Tayfun

**Comment by tenthe** [comment]
Hi @tayfun-celik,
Thanks for the information.

To retrieve the logs from the Docker container `extensions-all-jvm`, follow these steps:

1. Run `docker ps` to list all running containers and find the CONTAINER_ID of the `extensions-all-jvm` container.
2. Use `docker logs CONTAINER_ID`, replacing *CONTAINER_ID* with the actual ID, to print the logs to your terminal.
